agent/consul/server.go

```diff
@@ -898,8 +909,29 @@ func NewServer(config *Config, flat Deps, externalGRPCServer *grpc.Server,
 	return s, nil
 }
 
+func isV1CatalogRequest(rpcName string) bool {
+	switch {
+	case strings.HasPrefix(rpcName, "Catalog."),
+		strings.HasPrefix(rpcName, "Health."),
+		strings.HasPrefix(rpcName, "ConfigEntry."):
+		return true
+	}
+
+	switch rpcName {
+	case "Internal.EventFire", "Internal.KeyringOperation", "Internal.OIDCAuthMethods":
+		return false
+	default:
+		if strings.HasPrefix(rpcName, "Internal.") {
+			return true
+		}
+		return false
+	}
+}
+
 func (s *Server) registerControllers(deps Deps, proxyUpdater ProxyUpdater) error {
 	if stringslice.Contains(deps.Experiments, CatalogResourceExperimentName) {
+		s.useV2Resources = true
```

Comment:
this feels a little weird and the way we are parsing experiments in general is causing similar state to get set.  rather than try to do this via comments, I created this PR off of your branch that tries to centralize the useExperiments logic in Deps so that it is higher up than `agent` and can be used in agent as well as in `NewServer`.  Please take a look and merge into yours if it looks okay: https://github.com/hashicorp/consul/pull/19132